landside/catkin_ws/src/gui/src/gui/launch_dialog.py

```diff
@@ -102,49 +117,200 @@ def node_name_selected(self, item_index):
+        traverse_tree(root)
 
-            root = tree.getroot()
-            traverse_tree(root)
+    def interpret_options(self, selected_node, arg, doc_dict, input):
+        if (isinstance(doc_dict["options"], list) and doc_dict["options"] and
+                all(isinstance(e, str) for e in doc_dict["options"])):
 
-            for row in range(len(self.arg_form_rows)):
-                arg = self.arg_form_rows[row]
-                if arg.get('default') is None:
-                    default_value = ''
+            input = QtWidgets.QComboBox()
+            input.addItems(doc_dict["options"])
+
+            if arg.get("default") is not None:
+                if arg["default"] in doc_dict["options"]:
+                    input.setCurrentText(arg["default"])
                 else:
-                    default_value = arg['default']
+                    input.insertItem(0, arg["default"])
+                    input.setCurrentIndex(0)
+                    rospy.loginfo(f"Default value for argument `{arg['name']}` in `{selected_node}` "
+                                  f"is not in options list. It has been added as the first option.")
 
-                label = QtWidgets.QLabel(arg['name'])
-                input = QtWidgets.QLineEdit()
-                input.setText(default_value)
+        else:
+            rospy.logwarn(f"Options list for argument `{arg['name']}` in `{selected_node}` is not a "
+                          f"list of strings. Defaulting to unrestricted string input.")
 
-                # row inserted at position row+2, after the Package and Node Name rows
-                self.form_layout.insertRow(row + 2, label, input)
+        return input
 
-                arg['label'] = label
-                arg['lineEdit'] = input
+    def interpret_regex(self, selected_node, arg, doc_dict, input, toolTip):
+        try:
+            regex = QRegularExpression(doc_dict["regex"])
+            regex_validator = QtGui.QRegularExpressionValidator(regex)
+            if arg.get("default") is not None:
+                if regex.match(arg["default"]).hasMatch() or arg["default"] == "":
+                    input.setValidator(regex_validator)
+                    toolTip += f"Regex: {doc_dict['regex']}"
+                else:
+                    rospy.logwarn(f"Default value for argument `{arg['name']}` in `{selected_node}` "
+                                  f"does not match regex. Defaulting to unrestricted string input.")
+            else:
+                input.setValidator(regex_validator)
+                toolTip += f"Regex: {doc_dict['regex']}"
+
+        except Exception:
+            rospy.logwarn(f"Regex for argument `{arg['name']}` in `{selected_node}` is not valid. "
+                          f"Defaulting to unrestricted string input.")
+
+        return (input, toolTip)
+
+    def interpret_type(self, selected_node, arg, doc_dict, input, toolTip):
+        add_tooltip = True
+        if doc_dict["type"] == "bool":
```

Comment:
A `match` statement might make more sense here than this if-else chain

Reply:
`match` statements were introduced in Python 3.10. The Docker container and robot use Python 3.8.10.

Reply:
RIP